File: C05_PeerCast/PeerCastVB/PeerCast/BackgroundWorkerUtility.vb
Imports Microsoft.VisualBasic
Imports System.ComponentModel

Namespace PeerCast
	Public NotInheritable Class BackgroundWorkerUtility
        Public Shared Function Create() As BackgroundWorker
            Dim worker As New BackgroundWorker() With {.WorkerReportsProgress = True, .WorkerSupportsCancellation = True}
            Return worker
        End Function
	End Class
End Namespace

File: C5_PeerCast/PeerCastCS/PeerCast/BackgroundWorkerUtility.cs
using System.ComponentModel;

namespace PeerCast
{
    public static class BackgroundWorkerUtility
    {
        public static BackgroundWorker Create()
        {
            BackgroundWorker worker = new BackgroundWorker()            
                {
                    WorkerReportsProgress = true,
                    WorkerSupportsCancellation = true
                };
            return worker; 
        }
    }
}
